File: internal/content/testdata/bench/go/nc-edit-distance.go
func min_distance(word1 string, word2 string) int {
	m, n := len(word1), len(word2)
	dp := make([]int, n+1)
	for j := 0; j <= n; j++ {
		dp[j] = j
	}
	for i := 1; i <= m; i++ {
		prev := dp[0]
		dp[0] = i
		for j := 1; j <= n; j++ {
			temp := dp[j]
			if word1[i-1] == word2[j-1] {
				dp[j] = prev
			} else {
				minVal := prev
				if dp[j] < minVal {
					minVal = dp[j]
				}
				if dp[j-1] < minVal {
					minVal = dp[j-1]
				}
				dp[j] = 1 + minVal
			}
			prev = temp
		}
	}
	return dp[n]
}

File: internal/content/testdata/bench/java/nc-edit-distance.java
class Solution {
    public long min_distance(String word1, String word2) {
        int m = word1.length(), n = word2.length();
        long[] dp = new long[n + 1];
        for (int j = 0; j <= n; j++) dp[j] = j;
        for (int i = 1; i <= m; i++) {
            long prev = dp[0];
            dp[0] = i;
            for (int j = 1; j <= n; j++) {
                long temp = dp[j];
                if (word1.charAt(i - 1) == word2.charAt(j - 1)) {
                    dp[j] = prev;
                } else {
                    dp[j] = 1 + Math.min(prev, Math.min(dp[j], dp[j - 1]));
                }
                prev = temp;
            }
        }
        return dp[n];
    }
}
